app/src/org/commcare/activities/components/FormEntrySessionWrapper.java

```diff
@@ -0,0 +1,99 @@
+package org.commcare.activities.components;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+
+import org.commcare.activities.CommCareActivity;
+import org.commcare.views.UserfacingErrorHandling;
+import org.javarosa.core.util.externalizable.DeserializationException;
+import org.javarosa.core.util.externalizable.PrototypeFactory;
+import org.javarosa.form.api.FormEntrySession;
+import org.javarosa.form.api.FormEntrySessionReplayer;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+
+import static org.commcare.activities.FormEntryActivity.mFormController;
+
+/**
+ * Wrapper for the form entry action recorder, allowing for saving and
+ * restoring to the form entry activity.
+ *
+ * @author Phillip Mates (pmates@dimagi.com)
+ */
+public class FormEntrySessionWrapper {
```

Comment:
It contains Android specific code and `FormEntrySession` needs to live in `commcare-core`